crates/router/src/core/webhooks/outgoing.rs

```diff
@@ -136,6 +136,33 @@ pub(crate) async fn create_event_and_trigger_outgoing_webhook(
         is_overall_delivery_successful: Some(false),
     };
 
+    if !utils::perform_redis_lock(
+        &state,
+        idempotent_event_id.to_string().to_owned(),
+        merchant_account.get_id().to_owned(),
+    )
+    .await?
+    {
+        return Ok(());
+    }
+
+    if (state
+        .store
+        .find_event_by_merchant_id_event_id(
+            key_manager_state,
+            &merchant_id,
+            &event_id,
+            merchant_key_store,
+        )
+        .await)
+        .is_ok()
+    {
+        logger::debug!(
+            "Event with idempotent ID `{idempotent_event_id}` already exists in the database"
+        );
+        return Ok(());
+    }
+
     let event_insert_result = state
```

Comment:
I guess we can remove the `if error.current_context().is_db_unique_violation()` check we're performing in the following block.